akka-docs-dev/rst/scala/stream-graphs.rst

```diff
@@ -82,53 +92,59 @@ Constructing and combining Partial Flow Graphs
+Then we import it (all of its nodes and connections) explicitly into the closed graph built in the second step in which all
 the undefined elements are rewired to real sources and sinks. The graph can then be run and yields the expected result.
 
 .. warning::
+
    Please note that a :class:`FlowGraph` is not able to provide compile time type-safety about whether or not all
    elements have been properly connected - this validation is performed as a runtime check during the graph's instantiation.
 
+   A partial flow graph also verifies that all ports are either connected or part of the returned :class:`Shape`.
+
 .. _constructing-sources-sinks-flows-from-partial-graphs-scala:
 
 Constructing Sources, Sinks and Flows from Partial Graphs
 ---------------------------------------------------------
-Instead of treating a :class:`PartialFlowGraph` as simply a collection of flows and junctions which may not yet all be
+Instead of treating a partial flow graph as simply a collection of flows and junctions which may not yet all be
 connected it is sometimes useful to expose such a complex graph as a simpler structure,
 such as a :class:`Source`, :class:`Sink` or :class:`Flow`.
 
 In fact, these concepts can be easily expressed as special cases of a partially connected graph:
 
-* :class:`Source` is a partial flow graph with *exactly one* :class:`UndefinedSink`,
-* :class:`Sink` is a partial flow graph with *exactly one* :class:`UndefinedSource`,
-* :class:`Flow` is a partial flow graph with *exactly one* :class:`UndefinedSource` and *exactly one* :class:`UndefinedSource`.
+* :class:`Source` is a partial flow graph with *exactly one* output, that is it returns a :class:`SourceShape`.
+* :class:`Sink` is a partial flow graph with *exactly one* input, that is it returns a :class:`SinkShape`.
+* :class:`Flow` is a partial flow graph with *exactly one* input and *exactly one* output, that is it returns a :class:`FlowShape`.
 
 Being able to hide complex graphs inside of simple elements such as Sink / Source / Flow enables you to easily create one
 complex element and from there on treat it as simple compound stage for linear computations.
 
 In order to create a Source from a partial flow graph ``Source`` provides a special apply method that takes a function
-that must return an ``UndefinedSink``. This undefined sink will become "the sink that must be attached before this Source
-can run". Refer to the example below, in which we create a Source that zips together two numbers, to see this graph
+that must return an :class:`Outlet[T]`. This unconnected sink will become “the sink that must be attached before this Source
+can run”. Refer to the example below, in which we create a Source that zips together two numbers, to see this graph
 construction in action:
 
 .. includecode:: code/docs/stream/StreamPartialFlowGraphDocSpec.scala#source-from-partial-flow-graph
```

Comment:
these samples use explicit Flows as edges. Do we want to show the shorthand version where you chain from ports?